includes/class-wc-payments-woopay-direct-checkout.php

```diff
@@ -29,8 +29,42 @@ public function init() {
 	 * @return void
 	 */
 	public function scripts() {
+		// Only enqueue the script on the cart page, for now.
+		if ( ! $this->is_cart_page() ) {
+			return;
+		}
+
 		WC_Payments::register_script_with_dependencies( 'WCPAY_WOOPAY_DIRECT_CHECKOUT', 'dist/woopay-direct-checkout' );
 
+		$direct_checkout_settings = [
+			'params' => [
+				'is_product_page' => $this->is_product_page(),
+			],
+		];
+		wp_localize_script(
+			'WCPAY_WOOPAY_DIRECT_CHECKOUT',
+			'wcpayWooPayDirectCheckout',
+			$direct_checkout_settings
+		);
+
 		wp_enqueue_script( 'WCPAY_WOOPAY_DIRECT_CHECKOUT' );
 	}
+
+	/**
+	 * Check if the current page is the cart page.
+	 *
+	 * @return bool True if the current page is the cart page, false otherwise.
+	 */
+	public function is_cart_page(): bool {
+		return is_cart() || has_block( 'woocommerce/cart' );
+	}
+
+	/**
+	 * Check if the current page is the product page.
+	 *
+	 * @return bool True if the current page is the product page, false otherwise.
+	 */
+	public function is_product_page() {
+		return is_product() || wc_post_content_has_shortcode( 'product_page' );
```

Comment:
I may be mistaken but I don't believe there is a corresponding wc-blocks for the product page.

Reply:
I was referring to this one: https://github.com/woocommerce/woocommerce-blocks/blob/trunk/src/BlockTypes/SingleProduct.php

Reply:
Ah, I see. 

I don't think we would want to add the single-product block to this conditional because it is possible for a merchant to add a single-product block to their cart page. In the event that they did, they would be unknowingly removing the optimization.

What are your thoughts on this?

Reply:
Originally, that was my main concern but after reviewing the [docs for single-product](https://woo.com/document/woocommerce-blocks-configuration/#section-12), it looks like this won't be necessary because the single-product block only allows you to add the item to the cart (see image below).

![image](https://github.com/Automattic/woocommerce-payments/assets/6844516/4a872899-a9f2-4fe5-b454-badff48d8546)
[Image source](https://woo.com/wp-content/uploads/2023/01/Screenshot-2023-06-08-at-15.11.24.png?resize=1536,1016).